sklearn/base.py

```diff
@@ -335,6 +337,30 @@ def _get_tags(self):
         tags.update(collected_tags)
         return tags
 
+    def _validate_n_features(self, X, check_n_features):
+        if check_n_features:
+            if not hasattr(self, 'n_features_in_'):
+                raise RuntimeError(
+                    "check_n_features is True but there is no n_features_in_ "
+                    "attribute."
+                )
+            if X.shape[1] != self.n_features_in_:
+                raise ValueError(
+                    'X has {} features, but this {} is expecting {} features '
+                    'as input.'.format(X.shape[1], self.__class__.__name__,
+                                       self.n_features_in_)
+                )
+        self.n_features_in_ = X.shape[1]
+
+    def _validate_X(self, X, check_n_features=False, **check_array_params):
```

Comment:
Does using ``**kwargs`` make it harder to add and remove parameters to this function?
Seems potentially a bit tricky, right?
I guess the worst thing that could happen is that we add another argument here in a future version and a user using an older version trying to pass the argument will have it land in the ``**check_array_params`` and then ``check_array`` will raise an error?

Is there an issue with making all parameters explicit here?

Reply:
I'm not sure I understand your concern.

I can definitely make all the parameters explicit. The only downside is that we have to keep the signature synchronized with that of `check_array`, but that might be a good thing.

Reply:
That would only mean the user hasn't properly specified the minimum sklearn version requirement properly. I don't think that's something we should worry about.